Mostly zero change cleanups and stylings

SzieberthAdam opened the pull request.

SzieberthAdam committed:
Zerochange order of methods and styleing in FumbblApi.cs

SzieberthAdam committed:
Remove tailing curly brackets from certain roll modifier definitions

SzieberthAdam committed:
Add an empty line for more readabiltiy to AbstractModifier.cs

SzieberthAdam committed:
Color of Coach now follow Team and Player

SzieberthAdam committed:
Gender Nominative setter made as private to match Genetive and Dative

SzieberthAdam committed:
Zero change order of the Dative, Genetive, and Nominative fields

SzieberthAdam committed:
Merge remote-tracking branch 'upstream/master'

SzieberthAdam committed:
Zero change ordering in MonoBehaviour classes' modules

SzieberthAdam committed:
explicit access modes

SzieberthAdam committed:
remove unnecessary using directives

SzieberthAdam committed:
generalize ModelChange class code style

SzieberthAdam committed:
remove some more unnecessary using directives

SzieberthAdam committed:
add/remove empty lines for better readability

SzieberthAdam committed:
add Model Types using for PushBackSquare Coordinates to match practice

SzieberthAdam committed:
put constructor on top to match practice

SzieberthAdam committed:
zero change sort fields

SzieberthAdam committed:
maker Pass Block Log Text to use settings color

SzieberthAdam committed:
some more zero change code styling

christerk commented on Assets/ActionInjectorHandler.cs:
```diff
@@ -1,15 +1,19 @@
+
+
+    ///////////////////////////////////////////////////////////////////////////
+    //  MONOBEHAVIOUR METHODS  ////////////////////////////////////////////////
```
No, this is not how to do this.

If you're doing grouping of code like this, use `#region`

christerk commented on Assets/ActionInjectorHandler.cs:
```diff
@@ -35,10 +39,14 @@ void Update()
+
+
+    ///////////////////////////////////////////////////////////////////////////
+    //  CUSTOM METHODS  ///////////////////////////////////////////////////////
```
`#region`

SzieberthAdam commented:
Mostly zero change code uniformization and cleanups. I looked over the whole codebase multiple times for this. I really think that readability got better and the code look nicer after these changes. Tested with the Unity Editor.

Exceptions (minor real changes):
- bc4bf88: Coach color uses settings
- 946dd92: Gender Nominative field got to private to match other fields
- 1280706: added a using of Model.Types
- abde896: Pass Block color uses settings

christerk commented on Assets/BlockDiceHandler.cs:
```diff
@@ -1,32 +1,32 @@
 {
+    private ViewObjectList<BlockDie> BlockDice;
+    private static Vector2 FullSize = new Vector2(40, 40);
+    private static Vector2 SmallSize = new Vector2(30, 30);
```
If you are going to organize fields and properties, public before private is the way to go.

christerk commented on Assets/BlockDiceHandler.cs:
```diff
@@ -1,32 +1,32 @@
-    private static Vector2 SmallSize = new Vector2(30, 30);
-    private static Vector2 SpacerSize = new Vector2(12, 30);
+    ///////////////////////////////////////////////////////////////////////////
+    //  MONOBEHAVIOUR METHODS  ////////////////////////////////////////////////
```
`#region`. Not adding further comments about this particular style.

christerk commented on Assets/FieldHandler.cs:
```diff
@@ -1,48 +1,70 @@
 
 public class FieldHandler : MonoBehaviour
 {
+    private GameObject Ball;
```
public before private

christerk commented on Assets/FieldHandler.cs:
```diff
@@ -1,48 +1,70 @@
-    private ViewObjectList<Player> Players;
-    private ViewObjectList<PushbackSquare> PushbackSquares;
-    private ViewObjectList<TrackNumber> TrackNumbers;
+    private void OnDestroy()
```
I want life cycle methods grouped together. Start() followed by OnDestroy() to make it easier to follow when you read the code.

christerk commented on Assets/Scripts/Ffb/Dto/ModelChanges/FieldModelAddPushbackSquare.cs:
```diff
@@ -4,6 +4,8 @@ public class FieldModelAddPushbackSquare : ModelChange
         public PushbackSquare modelChangeValue;
 
-        public FieldModelAddPushbackSquare() : base("fieldModelAddPushbackSquare") { }
+        public FieldModelAddPushbackSquare() : base("fieldModelAddPushbackSquare")
```
This is unnecessary. The constructor calls the base class constructor and doesn't do anything and won't ever do anything beyond that. Leave it as it was.

christerk commented on Assets/Scripts/Model/Types/Player.cs:
```diff
@@ -5,28 +5,47 @@ namespace Fumbbl.Model.Types
-        public string Name;
-        public string PortraitURL;
-        public Types.Coordinate Coordinate { get; set; }
+        public Player()
```
As a general rule, fields and properties before constructor. I know the DTOs have the constructor first and if you insist on unifying this, it's better to change the DTOs to match the more general case.

christerk commented on Assets/SettingsHandler.cs:
```diff
@@ -83,46 +112,43 @@ public void ShowPanel(GameObject panel)
     }
 
-    public void Quit()
+    public void UpdateAbstractIcons()
```
public before private if you're reorganizing methods.

christerk commented on Assets/GameBrowserEntry.cs:
```diff
@@ -1,24 +1,41 @@
+
+
+    ///////////////////////////////////////////////////////////////////////////
+    //  CUSTOM METHODS  ///////////////////////////////////////////////////////
```
Don't need to state something as generic as "custom methods". Use `#region` to group monobehaviour and leave these outside that region.

christerk reviewed (changes requested):
This is a way too big PR. If you're doing this type of style cleanup, do separate PRs for each type of change to keep things small.

Also, do not have actual code changes in a PR like this as it becomes impossible to track those when looking back at commit history.

SzieberthAdam commented:
I give further comments to the individual commits to help reviewing:
- 7adbd14: Zero change. Methods sorted alphabetically.
- 483f9b1: Zero change. Only removed trailing ` {}`s. 
- 593ca1b: Zero change. Added an empty line.
- bc4bf88: **Minor change**: Coach color uses settings.
- 946dd92: **Minor change**: Gender Nominative field got to private to match other fields.
- 6e63d63: Zero change. Fields ordered alphabetically.
- 99dc591: Zero change. I expected MonoBehaviour subclasses' got their methods and fields just throwed in. Made some order and visually separated the custom and the mono methods.
- 7c87d5c: **Minor change**: Some access modes were explicit some were excluded. I made all of them explicit here.
- 360dc4f **Minor change**: The code were full of unnecessary using directives. I removed them where the IDE indicated it.
- e607c5c: Zero change. The ModelChange class constructors had their curly brackets put differently. Some were at the end like ` {}` and some were put into the subsequent two lines. I made all to the latter style.
- b7b45ed **Minor change**. Using directives again. Missed to click to stage them when I commited 360dc4f,
- fc382ea Zero change: Add and remove empty lines.
- 1280706 **Minor change**: added a using of Model.Types and chenged the qualified name of Coordinates
- 0cdd824 Zero change. Moved the constructor up to match other modules approach.
- b36def2 Zero change. Ordered the fields.
- abde896 **Minor change**: Pass Block Log text uses settings color
- ae35902 Zero change: Various. Missed to save them in the editor in the first place.

SzieberthAdam committed:
Make #regions for MonoBehaviour subclasses

SzieberthAdam committed:
MonoBehaviour subclass #regions done

SzieberthAdam committed:
Reorder of MonoBehaviour Methods: Awake, Start, Update, OnXY

SzieberthAdam committed:
Remove two empty lines between Start and Update methods

SzieberthAdam committed:
make ModelChanges constructor block simple empty

SzieberthAdam committed:
public before internal before private & some empty lines

SzieberthAdam committed:
public before private in PlayerState.cs

SzieberthAdam committed:
public, internal, and private, alphabetical order

SzieberthAdam committed:
remove of empty lines

SzieberthAdam committed:
remove of empty else block

SzieberthAdam committed:
constructors after fields and properties

SzieberthAdam commented:
Number of changed files has been grown as I moved the constructors of the DTOs after the fields.

SzieberthAdam committed:
remove of empty line

christerk commented:
Resolve the conflict please.

SzieberthAdam committed:
Make Login method public to mathc upstream

SzieberthAdam commented:
Resolved. LiginHandler.Login() got public in the meantime to be able to get called by the login scene button.

SzieberthAdam committed:
Merge branch 'master' into master
